src/functions.jl

```diff
@@ -138,3 +138,52 @@ function Base.append!(A::NamedDimsArray{L,T,1}, B::AbstractVector) where {L,T}
     data = append!(parent(A), unname(B))
     return NamedDimsArray{newL}(data)
 end
+
+################################################
+# map, collect
+
+# no explicit method needed for map(f, ::NamedDimsArray)
+for (T, S) in [
+    (:NamedDimsArray, :AbstractArray),
+    (:AbstractArray, :NamedDimsArray),
+    (:NamedDimsArray, :NamedDimsArray),
+    ]
+    for fun in [:map, :map!]
```

Comment:
No need to use symbols for these; you can use the functions directly

Reply:
> No need to use symbols for these; you can use the functions directly

Are you sure? 
```
julia> for fun in [sum, prod]
           @eval Base.$fun(x::String) = "yes"
       end
ERROR: syntax: invalid function name "Base.typeof(Base.sum)()"
Stacktrace:
 [1] top-level scope at REPL[14]:1
 [2] eval(::Module, ::Any) at ./boot.jl:331
 [3] top-level scope at REPL[14]:2

julia> sum("test")
ERROR: MethodError: no method matching +(::Char, ::Char)

julia> for fun in [:sum, :prod]
           @eval Base.$fun(x::String) = "yes"
       end

julia> sum("test")
"yes"
```

Reply:
Nvm you're right, I'm thinking of the eval-less method which you can't do with type parameters in variables.

Reply:
OK, but `x::String` has no parameters?